sdk/src/compressed_account.rs

```diff
@@ -262,67 +157,41 @@ where
 {
     input_account: T,
```

Comment:
if you have the state in the output account there is no need to keep it in the input account as well.
Instead you could just hash the data immediately in the beginning of the transaction and just save the `data_hash` in this field. Then you could assemble the complete compressed input account for example in the verify function.

Reply:
Done